rtengine/iplocallab.cc

```diff
@@ -10636,6 +10645,78 @@ void ImProcFunctions::avoidcolshi(struct local_params& lp, int sp, LabImage * or
                 }
             }
         }
+
+        //Guidedfilter to reduce artifacts in transitions
+        if (softr != 0.f) {//soft for L a b because we change color...
+            const float tmpblur = softr < 0.f ? -1.f / softr : 1.f + softr;
+            const int r1 = rtengine::max<int>(4 / sk * tmpblur + 0.5f, 1);
+            const int r2 = rtengine::max<int>(25 / sk * tmpblur + 0.5f, 1);
+
+            constexpr float epsilmax = 0.005f;
+            constexpr float epsilmin = 0.00001f;
+
+            constexpr float aepsil = (epsilmax - epsilmin) / 100.f;
+            constexpr float bepsil = epsilmin;
+            const float epsil = softr < 0.f ? 0.001f : aepsil * softr + bepsil;
+
+            const int bw = transformed->W;
+            const int bh = transformed->H;
+            array2D<float> ble(bw, bh);
+            array2D<float> guid(bw, bh);
+
+#ifdef _OPENMP
+            #pragma omp parallel for schedule(dynamic,16) if (multiThread)
+#endif
+
+            for (int y = 0; y < bh ; y++) {
+                for (int x = 0; x < bw; x++) {
+                    ble[y][x] = transformed->L[y][x] / 32768.f;
+                    guid[y][x] = original->L[y][x] / 32768.f;
+                }
+            }
+            rtengine::guidedFilter(guid, ble, ble, r2, 0.2f * epsil, multiThread);
+#ifdef _OPENMP
+            #pragma omp parallel for schedule(dynamic,16) if (multiThread)
+#endif
+            for (int y = 0; y < bh; y++) {
+                for (int x = 0; x < bw; x++) {
+                    transformed->L[y][x] = 32768.f * ble[y][x];
+                }
+            }
+
+            array2D<float> &blechro = ble; // reuse buffer
+#ifdef _OPENMP
+            #pragma omp parallel for schedule(dynamic,16) if (multiThread)
+#endif
+
+            for (int y = 0; y < bh ; y++) {
+                for (int x = 0; x < bw; x++) {
+                    blechro[y][x] = std::sqrt(SQR(transformed->b[y][x]) + SQR(transformed->a[y][x])) / 32768.f;
```

Comment:
Maybe using [`std::hypot`](https://en.cppreference.com/w/cpp/numeric/math/hypot) here and for `Chprov1` helps? (The description suggests better accuracy, but performance could be worse if the square root instruction isn't used.)